eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTests.java

```diff
@@ -0,0 +1,363 @@
+package com.netflix.discovery.util;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.shared.Application;
+import com.netflix.discovery.shared.Applications;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class EurekaEntityFunctionsTests {
+
+    @Test
+    public void testSelectApplicationNamesIfNotNullReturnLastNameAdded() {
+        Applications applications = new Applications();
+        applications.addApplication(new Application("foo"));
+        applications.addApplication(new Application("bar"));
+        applications.addApplication(new Application("baz"));
+
+        HashSet<String> strings = new HashSet<>();
```

Comment:
nit: reduce the amount of boiletprate by doing something like
`new HashSet<>(Arrays.asList("baz", "bar", "foo"));`, throughout where applicable.

Reply:
Done